src/hooks/useFavouriteMessages.ts

```diff
@@ -14,30 +14,61 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { useState } from "react";
+import { MatrixEvent } from "matrix-js-sdk/src/models/event";
+import { useCallback, useEffect } from "react";
 
-const favouriteMessageIds = JSON.parse(localStorage?.getItem("io_element_favouriteMessages") ?? "[]") as string[];
+import { FavouriteMessagesStore, FavouriteStorage } from "../stores/FavouriteMessagesStore";
 
-export default function useFavouriteMessages(): {
-    toggleFavourite: (eventId: string) => void;
+export default function useFavouriteMessages(favouriteMessageStore = FavouriteMessagesStore.instance): {
+    getFavouriteMessages: () => FavouriteStorage[];
     isFavourite: (eventId: string) => boolean;
+    toggleFavourite: (mxEvent: MatrixEvent) => void;
+    clearFavouriteMessages: () => void;
+    registerFavouritesChangedListener: (listener: () => void) => void;
 } {
-    const [, setX] = useState<string[]>();
+    const myListeners: (() => void)[] = [];
```

Comment:
At every render, `myListeners` is replaced by an empty array which seems to be unwanted behaviour. 

It needs to be a `useState` or `useRef`